plugins/advisors/black-duck/src/funTest/kotlin/ResponseCachingComponentServiceClient.kt

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (C) 2024 The ORT Project Authors (see <https://github.com/oss-review-toolkit/ort/blob/main/NOTICE>)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.plugins.advisors.blackduck
+
+import com.google.gson.GsonBuilder
+
+import com.synopsys.integration.blackduck.api.generated.response.ComponentsView
+import com.synopsys.integration.blackduck.api.generated.view.OriginView
+import com.synopsys.integration.blackduck.api.generated.view.VulnerabilityView
+
+import java.io.File
+import java.util.concurrent.ConcurrentHashMap
+
+/**
+ * This ComponentServiceClient uses a file backed cache for the responses.
+ * So, if the cache contains all responses for a particular test, an instance of this class can be used as a fake
+ * ComponentServiceClient.
+ */
+internal class ResponseCachingComponentServiceClient(
+    private val cacheFile: File,
+    private val serverUrl: String?,
+    apiToken: String?
+) : ComponentServiceClient {
+    // The BlackDuck library uses GSON to serialize its POJOs. So use GSON too because this is the simplest option.
```

Comment:
But you should not rely on Gson to be a transitive dependency, but add it explicitly to Gradle.